Added Bahasa Indonesia (id_ID) language support

evan-william opened the pull request:
This pull request introduces full support for the Indonesian language in Thonny by:
- Adding 'id_ID' to the LANGUAGES_DICT to enable language selection via the UI.
- Including the necessary translation files (.po and compiled .mo) under the locale/id_ID/LC_MESSAGES/ directory.

These changes aim to make Thonny more accessible for my fellow Indonesian-speaking users!

evan-william committed:
Add Bahasa Indonesia (id_ID) language support

aivarannamaa commented:
Thanks! Did you write these translations in Poeditor (see https://github.com/thonny/thonny/blob/master/docs/translate.md#short-story-for-translators) ?

Wowieee4 commented:
Thanks! I didn't use Poeditor , i manually edited the translation file after extracting them with Babel. Let me know if you'd prefer them doen through Poeditor instead.

aivarannamaa commented:
Please try to import your work into Poeditor. If this is not possible, then I'll accept the PR as is.

evan-william committed:
Updated thonny.po 

Updated the (PR) for Thonny's PO by replacing the previous file with the latest version exported from POEditor!

Wowieee4 commented:
Hi!, I've updated the file as requested. I've already imported the po file into POEditor and edited the thonny.po file from the PR to match the exported version from POEditor. Please let me know if this is what you meant. Thanks!

aivarannamaa commented:
Thanks!

The translations taken from Poeditor are now published in https://github.com/thonny/thonny/releases/tag/v5.0.0b1

evan-william commented:
Hi @aivarannamaa, thanks for the v5.0.0b1 release! 

I just checked the CREDITS.rst file and the release notes, but I didn't see my name there. Just wanted to double-check if my translation actually made it in?

I also noticed that the thonny.po file in the 5.0 branch seems to be an older/incomplete version (basic words like Cut/Copy/Paste are still in English), not the one I imported. Also, the POEditor link gives me a 'Not Found' error, probably because I don't have access to the project dashboard.

Let me know if there's anything else I need to do on POEditor to get my strings approved and my name on the credits, or if I'm just looking at the wrong branch or something. 

Cheers!

aivarannamaa commented:
I added you to CREDITS.rst. 

> the POEditor link gives me a 'Not Found' error, probably because I don't have access to the project dashboard.

Please try registering at poeditor and then try again.

evan-william commented:
Awesome, thank you so much for adding me to the credits!. Appreciate your help!
